client/me/purchases/remove-purchase/index.jsx

```diff
@@ -477,16 +455,26 @@ class RemovePurchase extends Component {
 			return null;
 		}
 
-		const purchase = this.props.purchase;
+		const { purchase, translate } = this.props;
+		const productName = getName( purchase );
+
 		if ( ! isRemovable( purchase ) ) {
 			return null;
 		}
 
 		return (
-			<span>
-				{ this.renderCard() }
+			<Fragment>
+				<CompactCard
+					tagName={ Focusable }
```

Comment:
> Warning: Failed prop type: Invalid prop `tagName` of type `function` supplied to `Card`, expected `string`.

Reply:
Try `tagName="Focusable"` maybe?

Reply:
Hmm no, that seems to break focusing by tabbing.

So the workaround that immediately comes to mind would be to allow for components as `propTypes` in `Card`.

However, taking a step back: Shouldn't a `Card` always be focusable? I.e. do we actually need to change the implementation of `Card` accordingly?

Reply:
Bad `propType`, bad!

Reply:
Fixed the bad propType in 2316a25.

I don't think cards aren't necessarily focusable. It might make sense to make the default type `Focusable` if they have click handlers but not hrefs… but maybe out of scope here.

Reply:
> It might make sense to make the default type Focusable if they have click handlers

That sounds like a good enhancement for another PR definitely 🙂  But cards with `href` should stay `<a />`s, and cards without interactions should stay `<div />`s.

When `Focusable` was created, I actually wondered about allowing a tagName prop, since I thought focusable `Card`s might be common -- I think this suggestion (of using Focusable in Card) would be a better way to handle that, though 👍

Reply:
> When `Focusable` was created, I actually wondered about allowing a tagName prop, since I thought focusable `Card`s might be common -- I think this suggestion (of using Focusable in Card) would be a better way to handle that, though 👍 

The tricky part is how to retain `Card`'s feature to accept a `tagName` prop. I guess we could add the ARIA `button` role and specialised `onKeyDown` handling directly to the `tagName` component. That would still allow some `tagName` components that would be unsemantic in terms of a11y, but our ESLint rules wouldn't catch them since we're sneaking in `tagName` dynamically... :thinking:

Reply:
Hm, it actually looks like [the domains section](https://github.com/Automattic/wp-calypso/blob/master/client/components/domains/register-domain-step/index.jsx#L471-L483) solves all of this by passing in `tagName="button"` which would be a better solution for this instance - it's simpler, and more semantic.

I'm not sure how to best address this for other possibly-clickable cards, since it's so dependent on the content of the card.

Reply:
> […] [the domains section](https://github.com/Automattic/wp-calypso/blob/master/client/components/domains/register-domain-step/index.jsx#L471-L483) solves all of this by passing in `tagName="button"` which would be a better solution for this instance - it's simpler, and more semantic.

Sorry, I'm a bit lost if we're talking about this PR or the Card enhancements…

I'd initially used the `tagName="button"` approach, but it screwed up styles so I went with this. Happy to take that approach if its preferred. Here's what the `button` approach looks like without special styling:

![button](https://user-images.githubusercontent.com/841763/41096128-f41820f8-6a53-11e8-9da9-1c6931466e2c.png)